### ISSUE TITLE
Use vctrs::s3_register()

### ISSUE THREAD

**Issue by krlmlr** [r-lib/pillar#155]
In a separate issue/PR, to isolate potential problems.

Blocked by #152.

**Comment by krlmlr** [comment]
Need to wait for {vctrs} > 0.1.0 on CRAN.

**Comment by krlmlr** [comment]
Can't use, see `.onLoad`.
